llvm/lib/DebugInfo/LogicalView/Readers/LVBinaryReader.cpp

```diff
@@ -433,6 +433,13 @@ Error LVBinaryReader::createInstructions(LVScope *Scope,
 
   ArrayRef<uint8_t> Bytes = arrayRefFromStringRef(*SectionContentsOrErr);
   uint64_t Offset = Address - SectionAddress;
+  if (Offset > Bytes.size()) {
+    LLVM_DEBUG({
+      dbgs() << "offset (" << hexValue(Offset) << ") is beyond section size ("
+             << hexValue(Bytes.size()) << "); malformed input?\n";
+    });
+    return Error::success(); // Continue; returning error aborts reader.
+  }
```

Comment:
I think we should handle this condition as an error and abort the reader. May be something like:
```
if (Offset > Bytes.size()) {
  LLVM_DEBUG({.....});
  return createStringError(errc::bad_address, .......);
}
```

Reply:
Fixed!  Please, take a look.  I have also rebased atop current `main` state.

I think this is in mergeable state now.  Also, I believe it should not cause any merging conflicts with https://github.com/llvm/llvm-project/pull/135440.